content/en/docs/setup/install/multicluster/virtual-machine/index.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Installing Istio for virtual machine connectivity
+description: Install an Isito mesh such that a virtual machine can connect to it and configure the virtual machine.
+weight: 3
+keywords:
+- kubernetes
+- virtual-machine
+- gateways
+- vms
+---
+
+Follow this guide to install an Istio control plane
+[multicluster deployment](/docs/ops/deployment/deployment-models/#multiple-clusters) and connect your first virtual machine.
+
+## Prerequisites
+
+- One or more Kubernetes clusters with versions: {{< supported_kubernetes_versions >}}.
+
+- Authority to [deploy the Istio control plane](/docs/setup/install/istioctl/).
+
+- Virtual machines must have IP connectivity to the ingress gateway in the connecting mesh, or alternatively, every pod in the mesh via L3 networking.
```

Comment:
Environments wishes to remove this alternative L3 model. See https://github.com/istio/istio.io/issues/7168 for more detial.

Reply:
The approach SGTM! A headless service would work if the VM used DNS from the K8s istiod was running in. I am not sure how it could work any other way. I agree with the approach. Removing memcpy() is an essential performance enhancement in any software.

I am not sure that the implementation is there yet, although happy to document what is present.

@costinm any thoughts? I know you sorted out the DNS mapping with coredns, although I don't know what the workflow for that looks like inside a VM.

Reply:
@smawson after having slept on it, I understand what you are after. I have corrected the language of the requirement.

As such, resolving this comment.

Reply:
The L3 is required for workloads to talk with each other. So headless istiod doesn't help, it wouldn't be reachable.  

For DNS - there are several options for VM:
- create a LB Service for the K8S DNS  UDP port and set it in resolv.conf (simple, insecure)
- run CoreDNS on the VM - configured to foward to Istiod DNS-TLS port ( or to a DNS-TLS port in the gateway that forwards to Kube DNS)
- use external DNS, and the kubernetes-sigs/external-dns to sync
- for simple cases with few outbound calls - just use etc/hosts

In theory 'multi network' should work for VMs as well - I don't think anyone tested,
but the code and design supports VMs as well.

Reply:
In other words: for 1.6 unless someone tested 'multi network' with VMs - I suggest keeping the language closer to reality.

Reply:
Multi-network instead of single network doesn't make much difference unless the L3 case is considered. I have exclusively tested multiple networks connected together without l3 connectivity. I have the capability to start K8s in GKE - although I have to personally cover the cost - to test out L3 connectivity. The L3 connectivity option would be a decision point about when to use DNS, and how to use DNS securely remains a topic for a follow-up document (you can see in this document, as well as all prior documents on this topic, we have punted on the topic of DNS).

I did test headless connected o a service, and resolving the headless service via DNS produces a list of pod IPs, so I really do think headless is an option here, but it would have to select with istiod and this isn't implemented ATM.

GIven the level of pain to merge the makefile change to enable all of this work, I'm not keen to do that for 1.6, but at least merge this basic essential documentation for 1.6.